### PR TITLE
Add support for update policy configuration

### LINKED ISSUE #229: Maven Artifact Resolver: Ability to provide more Remote repository properties
Currently, The MavenArtifact resolver uses only a set of properties to update the [RemoteRepositoryBuilder](https://github.com/spring-cloud/spring-cloud-deployer/blob/master/spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenArtifactResolver.java#L116) when creating/resolving the artifacts against the given remote repositories. 

We need to identify and update all the supported properties for the RemoteRepositoryBuilder.

For instance, setting the update policy for snapshots is one such an option.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add update policy for snapshot, release artifacts via remote repository configuration
 - Add common policy update to remote repository configuration
 - Add tests

Resolves #229

### CODE REVIEW

File: spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenArtifactResolver.java [hunk 1]
```diff
@@ -58,9 +58,10 @@
 import org.springframework.util.StringUtils;
 
 /**
- * Resolves a {@link MavenResource} using <a href="http://www.eclipse.org/aether/>aether</a> to
- * locate the artifact (uber jar) in a local Maven repository, downloading the latest update from a
- * remote repository if necessary.
+ * Resolves a {@link MavenResource} using <a
```
Comment (trisberg) [inline spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenArtifactResolver.java]: Did you change something here? Hard to tell with the reformatting

Comment (ilayaperumalg) [reply spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenArtifactResolver.java]: Sorry, no changes here.

**Comment by sabbyanandan** [comment]
@ilayaperumalg: ping

**Comment by trisberg** [comment]
LGTM except for some of the formatting of the comments - I can remove them on merge

**Comment by trisberg** [comment]
Merged as commit 1b40ff13778ae686b5f90aa913926f2fb03a779b

**Comment by ilayaperumalg** [comment]
@trisberg Thank you for removing the formatting changes.

**Pull request closed without merging**